execution_engine_testing/tests/benches/transfer_bench.rs

```diff
@@ -247,6 +250,67 @@ pub fn transfer_to_existing_accounts(group: &mut BenchmarkGroup<WallTime>, shoul
     );
 }
 
+pub fn multiple_native_transfers(group: &mut BenchmarkGroup<WallTime>) {
+    let target_account = TARGET_ADDR;
+    let bootstrap_accounts = vec![target_account];
+
+    let data_dir = TempDir::new().expect("should create temp dir");
+    let mut builder = bootstrap(
+        data_dir.path(),
+        bootstrap_accounts,
+        U512::from(MINIMUM_ACCOUNT_CREATION_BALANCE),
+    );
+
+    let purse_amount = U512::one();
+    let purses = create_purses(&mut builder, target_account, 100, purse_amount);
+
+    let mut purse_index = 0usize;
+    let mut exec_requests = Vec::with_capacity(BLOCK_TRANSFER_COUNT);
+    for _ in 0..BLOCK_TRANSFER_COUNT {
+        let account = {
+            let account = purses[purse_index];
+            if purse_index == purses.len() - 1 {
+                purse_index = 0;
+            } else {
+                purse_index += 1;
+            }
+            account
+        };
+        let mut exec_builder = ExecuteRequestBuilder::new();
+        let runtime_args = runtime_args! {
+            ARG_TARGET => account,
+            ARG_AMOUNT => U512::one(),
+            ARG_ID => <Option<u64>>::None
+        };
+        let native_transfer = DeployItemBuilder::new()
+            .with_address(*DEFAULT_ACCOUNT_ADDR)
+            .with_empty_payment_bytes(runtime_args! {})
+            .with_transfer_args(runtime_args)
+            .with_authorization_keys(&[*DEFAULT_ACCOUNT_ADDR])
+            .build();
+        exec_builder = exec_builder.push_deploy(native_transfer);
+        let exec_request = exec_builder.build();
+        exec_requests.push(exec_request);
+    }
+
+    group.bench_function(
+        format!("multiple_native_transfers/{}", BLOCK_TRANSFER_COUNT),
+        |b| b.iter(|| transfer_to_account_multiple_native_transfers(&mut builder, &exec_requests)),
+    );
+}
+
+fn transfer_to_account_multiple_native_transfers(
+    builder: &mut LmdbWasmTestBuilder,
+    execute_requests: &[ExecuteRequest],
+) {
+    for exec_request in execute_requests.iter().cloned() {
+        let builder = builder.exec(exec_request).expect_success();
```

Comment:
This would allow us to avoid making the `ExecuteRequest` clonable:
```suggestion
    for exec_request in execute_requests {
        let request = ExecuteRequest::new(
            exec_request.parent_state_hash,
            exec_request.block_time,
            exec_request.deploys.clone(),
            exec_request.protocol_version,
            exec_request.proposer.clone(),
        );
        let builder = builder.exec(request).expect_success();
```